paderbox/utils/random_utils.py

```diff
@@ -37,59 +37,90 @@ def str_to_random_state(string):
     return np.random.RandomState(hash_value)
 
 
-def _force_correct_shape(f):
-    """decorator allowing to pass shape as tuple as well
+def _force_correct_shape(shape):
+    if len(shape) > 0 and isinstance(shape[0], (tuple, list)):
+        assert len(shape) == 1, shape
+        shape = shape[0]
+    return shape
 
-    Args:
-        f: wrapped function
 
-    Returns:
+@dataclasses.dataclass
+class _Sampler:
+    dtype: type = np.float64
 
-    """
-    @wraps(f)
-    def wrapper(*shape, **kwargs):
-        if len(shape) > 0 and isinstance(shape[0], (tuple, list)):
-            assert len(shape) == 1, shape
-            shape = shape[0]
+    def _sample(self, *shape):
```

Comment:
Why the asterisk expansion? This is a private method and it would work just fine if shape would be passed directly.